drools-ansible-rulebook-integration-api/src/main/java/org/drools/ansible/rulebook/integration/api/domain/conditions/MapCondition.java

```diff
@@ -239,8 +273,10 @@ private static RulebookOperator decodeOperation(String expressionName) {
             case "LessThanOrEqualToExpression":
                 return RulebookOperator.LESS_OR_EQUAL;
             case ExistsField.EXPRESSION_NAME:
-            case ExistsField.NEGATED_EXPRESSION_NAME:
                 return ExistsField.INSTANCE;
+            case ExistsField.NEGATED_EXPRESSION_NAME:
+                // IsNotDefinedExpression behaves as an existential not only when used alone
+                return ruleContext.getCondition().isSingleCondition() ? ExistsField.INSTANCE : NegatedExistsField.INSTANCE;
```

Comment:
Here I'm clearly making a distinction between the case when the `IsNotDefinedExpression` is used alone, when it is evaluated as our `not` existential operator, or in conjuction with another constraint, when it is simply the negation of `IsDefinedExpression`.